core/Session/SessionAuth.php

```diff
@@ -186,4 +192,24 @@ private function updateSessionExpireTime()
         setcookie(session_name(), session_id(), time() + $sessionCookieLifetime, $sessionParams['path'],
             $sessionParams['domain'], $sessionParams['secure'], $sessionParams['httponly']);
     }
+
+    private function isExpiredSession(SessionFingerprint $sessionFingerprint)
+    {
+        $nonRememberedSessionExpireTime = Config::getInstance()->General['login_session_not_remembered_idle_timeout'];
+        $sessionCookieLifetime = Config::getInstance()->General['login_cookie_expire'];
+
+        $startTime = $sessionFingerprint->getSessionStartTime();
+        if ($startTime === null) {
+            return false;
+        }
+
+        if ($sessionFingerprint->isRemembered()) {
+            $expireTime = $startTime + $sessionCookieLifetime;
+        } else {
+            $expireTime = $startTime + $nonRememberedSessionExpireTime;
```

Comment:
Here I'm seeing this warning quite often:

> WARNING [2019-06-11 08:15:17 UTC] [5b1e8 ] > core/Session/SessionAuth.php(209): Warning - A non-numeric value encountered - Matomo 3.10.0-b3 - Please report this message in the Matomo forums: https://forum.matomo.org (please do a search first as it might have been reported already)